packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -883,7 +883,13 @@ class _AppBarState extends State<AppBar> {
     final SystemUiOverlayStyle style = brightness == Brightness.dark
       ? SystemUiOverlayStyle.light
       : SystemUiOverlayStyle.dark;
-    return style.copyWith(statusBarColor: backgroundColor);
+    // For backward compatibility, create an overlay style without system navigation bar settings.
+    return SystemUiOverlayStyle(
+      statusBarColor: style.statusBarColor,
```

Comment:
Good catch! If we pursue work on this PR, I will also try to find why this bug does not break any tests.

Reply:
I fixed this bug and I added a test as there was no existing test covering this (it is somewhat new because the 'backgroundColor' parameter is non-null only when using Material 3). 
Thanks a lot for spotting this!